action_server/src/action_server/client.py

```diff
@@ -140,8 +140,9 @@ def send_task(self, semantics):
             raise KeyboardInterrupt
 
         if not isinstance(result, action_server_msgs.msg.TaskResult):
-            rospy.logerr("Result not instance of 'action_server_msgs.msg.TaskResult', but {}".format(type(result)))
-            return False
+            msg = "Result not instance of 'action_server_msgs.msg.TaskResult', but {}".format(type(result))
+            rospy.logerr(msg)
+            return TaskOutcome(messages=[msg])
```

Comment:
`:return: True or false, and a message specifying the outcome of the task`

Some type annotations also wouldn't hurt I guess in the end, but probably for another PR

Reply:
This doc is incorrect. It should return TaskOutcome. Check `task_outcome_from_result` for correct return type. I can fix the doc, though it is not really in the scope of the fix.